File: backend/src/main/java/com/stadium/booking/entity/Venue.java
package com.stadium.booking.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import java.time.LocalTime;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("venue")
public class Venue extends BaseEntity {
    private Long campusId;
    private String name;
    private String code;
    private String sportType;
    private String location;
    private String description;
    private String imageUrl;
    private String openDays;
    private LocalTime openTime;
    private LocalTime closeTime;
    private Integer slotMinutes;
    private Integer bookAheadDays;
    private Integer cancelCutoffMinutes;
    private Integer checkinWindowBefore;
    private Integer noShowGraceMinutes;
    private Integer dailySlotLimit;
    private Integer weeklySlotLimit;
    private Integer groupBookingEnabled;
    private Integer groupMaxCourts;
    private Integer groupMaxHours;
    private Integer status;
}

File: miniapp/src/types/venue.ts
export interface Venue {
  id: number
  name: string
  code?: string
  sportType: string
  location: string
  description?: string
  imageUrl?: string
  openDays?: string
  openTime: string
  closeTime: string
  slotMinutes?: number
  bookAheadDays?: number
  cancelCutoffMinutes?: number
  dailySlotLimit?: number
  status: number
  courtCount?: number
  courts?: Court[]
}

export interface Court {
  id: number
  venueId?: number
  venueName?: string
  name: string
  courtNo?: string
  sportType?: string
  floorType?: string
  features?: string
  status: number
  statusReason?: string
  statusUntil?: string
  sortOrder?: number
}

export interface TimeSlot {
  courtId?: number
  courtName?: string
  date?: string
  startTime: string
  endTime: string
  status: string
  booking?: BookingInfo
}

export interface BookingInfo {
  bookingNo: string
  userName: string
  userPhone: string
}
